iocage/lib/Provisioning/ix.py

```diff
@@ -0,0 +1,241 @@
+
+        ioc set provisioning.method=ix provisioning.source=jenkins myjail
+    """
+    events = iocage.lib.events
+    jailProvisioningEvent = events.JailProvisioning(
+        jail=self.jail,
+        event_scope=event_scope
+    )
+    yield jailProvisioningEvent.begin()
+    _scope = jailProvisioningEvent.scope
+    event_kargs = dict(
+        jail=self.jail,
+        event_scope=_scope
+    )
+    jailProvisioningAssetDownloadEvent = events.JailProvisioningAssetDownload(
+        **event_kargs
+    )
+    packageUpdateEvent = PackageUpdateEvent(
+        **event_kargs
+    )
+    packageInstallEvent = PackageInstallEvent(
+        **event_kargs
+    )
+    postInstallExecutionEvent = PostInstallExecutionEvent(
+        **event_kargs
+    )
+
+    # download provisioning assets
+    try:
+        yield jailProvisioningAssetDownloadEvent.begin()
+        pluginDefinition = PluginDefinition(
+            self.source,
+            logger=self.jail.logger
+        )
+        yield jailProvisioningAssetDownloadEvent.end()
+    except Exception as e:
+        yield jailProvisioningAssetDownloadEvent.fail(e)
+        raise e
+
+    # clone plugin
+    plugin_dataset_name = f"{self.jail.dataset.name}/ix-plugin"
+    plugin_dataset = __get_empty_dataset(plugin_dataset_name, self.jail.zfs)
+    git.Repo.clone_from(
+        pluginDefinition["artifact"],
+        plugin_dataset.mountpoint
+    )
+
+    self.jail.fstab.file = "fstab_provisioning"
+    self.jail.fstab.new_line(
+        source=plugin_dataset.mountpoint,
+        destination="/.ix-plugin",
+        options="ro",
+        auto_create_destination=True
+    )
+    self.jail.fstab.save()
+
+    if "pkgs" in pluginDefinition.keys():
+        pkg_packages = list(pluginDefinition["pkgs"])
+    else:
+        pkg_packages = [pluginDefinition.name]
+
+    for event in self.jail.start():
+        yield event
+
+    try:
+        yield packageUpdateEvent.begin()
+        try:
+            stdout, stderr, returncode = self.jail.exec(
+                ["pkg", "update"],
+                env=dict(ASSUME_ALWAYS_YES="true")
+            )
+            yield packageUpdateEvent.end(stdout=stdout)
+        except iocage.lib.errors.IocageException as e:
+            yield packageUpdateEvent.fail(e)
+            raise e
+
+        yield packageInstallEvent.begin()
+        try:
+            stdout, stderr, returncode = self.jail.exec(
+                ["pkg", "install", "-y"] + pkg_packages,
+                env=dict(ASSUME_ALWAYS_YES="true")
+            )
+            yield packageInstallEvent.end(stdout=stdout)
+        except iocage.lib.errors.IocageException as e:
+            yield packageInstallEvent.fail(e)
+            raise e
+
+        yield postInstallExecutionEvent.begin()
+        try:
+            if os.path.isfile(f"{plugin_dataset.mountpoint}/post_install.sh"):
+                stdout, stderr, returncode = self.jail.exec(
+                    ["/.ix-plugin/post_install.sh"]
```

Comment:
is this file guaranteed to be executable?

Reply:
It was created by ourselves. Yes, we can assume that this file is existing and executable.